src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs

```diff
@@ -76,296 +67,106 @@ public override async Task Dispatch(PeerInfo bestPeer, BlocksRequest? blocksRequ
-                                downloadReceipts = true;
-                                context.SetDownloadReceipts();
-                                await RequestReceipts(bestPeer, cancellation, context);
-                                receipts = context.ReceiptsForBlocks;
-                            }
-                        }
-                    }
-
-                    if (downloadReceipts)
-                    {
-                        TxReceipt[]? contextReceiptsForBlock = receipts![blockIndex];
-                        if (currentBlock.Header.HasTransactions && contextReceiptsForBlock is null)
-                        {
-                            throw new EthSyncException($"{bestPeer} didn't send receipts for block {currentBlock.ToString(Block.Format.Short)}.");
-                        }
-                    }
-
-                    bool isKnownBeaconBlock = _blockTree.IsKnownBeaconBlock(currentBlock.Number, currentBlock.GetOrCalculateHash());
-                    BlockTreeSuggestOptions suggestOptions =
-                        shouldProcess ? BlockTreeSuggestOptions.ShouldProcess : BlockTreeSuggestOptions.None;
-                    if (_logger.IsTrace)
-                        _logger.Trace(
-                            $"Current block {currentBlock}, BeaconPivot: {_beaconPivot.PivotNumber}, IsKnownBeaconBlock: {isKnownBeaconBlock}");
-
-                    if (isKnownBeaconBlock)
-                    {
-                        suggestOptions |= BlockTreeSuggestOptions.FillBeaconBlock;
-                    }
-
-                    if (_logger.IsTrace)
-                        _logger.Trace(
-                            $"MergeBlockDownloader - SuggestBlock {currentBlock}, IsKnownBeaconBlock {isKnownBeaconBlock} ShouldProcess: {shouldProcess}");
+                    _logger.Trace("Chain level helper got no headers suggestion");
+                return Task.FromResult<IOwnedReadOnlyList<BlockHeader?>?>(null);
+            }
 
-                    AddBlockResult addResult = _blockTree.SuggestBlock(currentBlock, suggestOptions);
-                    if (HandleAddResult(bestPeer, currentBlock.Header, blockIndex == 0, addResult))
-                    {
-                        if (shouldProcess == false)
-                        {
-                            _blockTree.UpdateMainChain(new[] { currentBlock }, false);
-                        }
-                        else
-                        {
-                            bestProcessedBlock = currentBlock.Number;
-                        }
+            // Alternatively we can do this in BeaconHeadersSyncFeed, but this seems easier.
+            ValidateSeals(headers!, cancellation);
 
-                        TryUpdateTerminalBlock(currentBlock.Header, shouldProcess);
+            return Task.FromResult<IOwnedReadOnlyList<BlockHeader?>?>(headers.ToPooledList(0));
```

Comment:
Why 0? that would be `ArrayPoolList.Empty`?

Reply:
Fixed